language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Function.hs

```diff
@@ -56,24 +57,24 @@ selfData = runQuote $ do
 -- | @unroll@ as a PLC term.
 --
 -- > /\(a :: *) -> \(s : self a) -> unwrap s s
-unroll :: Term TyName Name ()
+unroll :: forall term . TermLike term TyName Name => term ()
 unroll = runQuote $ do
-    let self = _recursiveType selfData
+    let self = _recursiveType (selfData :: RecursiveType term ())
```

Comment:
I'm kind of surprised you need these annotations...

Reply:
(I messed up and PR-d an older commit so this is fixed in HEAD)
This used to be necessary because selfData would have type `TermLike term ... => RecursiveType term ()` but `_recursiveType` doesn't mention `term` so you need to disambiguate `term` so the right typeclass instance can be used (although it's never *actually* needed)